compiler/rustc_passes/src/dead.rs

```diff
@@ -623,6 +623,11 @@ fn create_and_seed_worklist<'tcx>(
         check_foreign_item(tcx, &mut worklist, id);
     }
 
+    if let Some(static_) = tcx.proc_macro_decls_static(()) {
```

Comment:
Would adding`#[used]` when creating this static work?

Reply:
+1, if it's possible we should rely on regular user-available attributes instead of hardcoding stuff in the compiler, both for dead code and symbol export analysis.

Reply:
Interaction between dead code lint and #[used] attribute is tested in issue-41628.rs.